table/requirement_test.go

```diff
@@ -145,3 +147,214 @@ func TestParseRequirementList(t *testing.T) {
 		assert.Error(t, err)
 	})
 }
+
+// mockMetadata is a correct and complete mock implementation of the
+// table.Metadata interface, used for testing requirement validation.
+type mockMetadata struct {
```

Comment:
this is not a mock, this is fake, take a look [here](https://martinfowler.com/bliki/TestDouble.html)